packages/browser-data-collector/src/collectors/blocking-resources.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import { getResources } from '../api/resources-timing';
+import { getDomContentLoadedEventStart, getNavigationStart } from '../api/performance-timing';
+
+export const collector: Collector = ( report ) => {
+	const domContentLoaded = getDomContentLoadedEventStart() - getNavigationStart();
+	if ( domContentLoaded <= 0 ) return report;
+
+	/**
+	 * Fetch resources that
+	 * 	- are scripts
+	 *  - finish downloading before domContentLoaded
+	 *  - have a size (to exclude resources form a different origin)
+	 */
+	const blockingResources = getResources().filter(
+		( r ) =>
+			r.initiatorType === 'script' && r.responseEnd <= domContentLoaded && r.decodedBodySize > 0
+	);
+
+	const resourcesCount = blockingResources.length;
+	const resourcesStart = blockingResources.sort( ( a, b ) => a.requestStart - b.requestStart )[ 0 ]
+		.requestStart;
+	const resourcesEnd = blockingResources.sort( ( a, b ) => b.requestStart - a.requestStart )[ 0 ]
+		.responseEnd;
```

Comment:
If you want to figure out when the last response ended, then the second sorting should be by `.responseEnd` instead of `.requestStart`, right?

Anyway, sorting the array twice just to find a min/max value is somewhat inefficient. This is faster and even more concise:
```js
const resourcesStart = blockingResources.reduce( ( min, r ) => Math.min( min, r.requestStart ), Infinity );
const resourcesEnd = blockingResources.reduce( ( max, r ) => Math.max( max, r.responseEnd ), -Infinity );
```
The two `reduce` loops can further be reduced into one.

Reply:
That's much better, thanks!